Contributing.md

```diff
@@ -0,0 +1,124 @@
+A good introduction can be found [here](https://www.youtube.com/watch?v=P6jD966jzlk).
+
+Following this, I would highly recommend reading [this](https://www.atlassian.com/git/tutorials/cherry-pick) article on what cherry picking is, and [this](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase) on rebasing.
+Altassian's documentation in general is quite good if you're looking for how some command is meant to be used.
+
+### Git workflow
+[This](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow) document provides a pretty good overview of the "feature branch" workflow, which gives a pretty good high-level overview of the git workflow.
+It should be noted, however, that feature branch editing tends to imply that you'll finish an entire feature before putting it up for code review.
+Instead, just get to the point where you have passing tests for as small an amount of changes as is reasonable, and hopefully a single commit, and then put in a pull request.
+I would recommend naming whatever branch you are working on with just your name; it saves cognitive overhead.
+
+### Commit Hygiene
+#### Commit Messages
+Read [this](https://chris.beams.io/posts/git-commit/) piece on the matter.
+If you're feeling particularly short on time skim the first couple paragraphs and then jump down to the "rules". To reiterate them, though:
+
+ - Separate subject from body with a blank line
+ - Limit the subject line to 50 characters
+ - Capitalize the subject line
+ - Do not end the subject line with a period
+ - Use the imperative mood in the subject line
+ - Wrap the body at 72 characters
+ - Use the body to explain what and why vs. how
+
+If you're looking for how to style things, use [markdown](https://www.markdownguide.org/cheat-sheet/) syntax.
+(That's also how this file is written)
+
+As an aside, using the `git commit -m "My commit message"` syntax to write commit messages is somewhat limiting, and I would recommend that you use the default `git commit` command, which simply brings up a text editor to edit the message with.
+To change this editor on MacOS and Linux, you want to set the `EDITOR` and `VISUAL` environment variables to the name of the text editors that you want to use.
+For example, if you want to use `nano`, you would write the following:
+
+    export EDITOR='nano'
+    export VISUAL='nano'
+
+To keep these changes when you re-open your shell, you should add these two lines to your `~/.bashrc`.
+
+To do this on windows, see [this](https://stackoverflow.com/questions/2114111/where-does-git-config-global-get-written-to) entry on stackoverflow.
+
+#### Commit size
+The big thing to keep in mind is that you really don't want to make much progress without putting in a pull request.
+Try to keep each pull request to a single commit, doing a single change.
+As a rough metric; 80 lines of code added is probably too many, as a rough estimate.
```

Comment:
Grammatical thing. "as a rough estimate" is redundant after starting the sentence with "As a rough metric."

Reply:
Thanks! Resolved.